clientconn.go

```diff
@@ -1676,13 +1675,25 @@ func parseTarget(target string) (resolver.Target, error) {
 	if err != nil {
 		return resolver.Target{}, err
 	}
-	path := u.Path
-	if path == "" {
-		path = u.Opaque
+	// For targets of the form "[scheme]://[authority]/endpoint, the endpoint
+	// value returned from url.Parse() contains a leading "/". Although this is
+	// in accordance with RFC 3986, we do not want to break existing resolver
+	// implementations which expect the endpoint without the leading "/". So, we
+	// end up stripping the leading "/" here. But this will result in an
+	// incorrect parsing for something like "unix:///path/to/socket". Since we
+	// own the "unix" resolver, we can workaround in the unix resolver by using
+	// the `ParsedURL` field instead of the `Endpoint` field.
+	endpoint := u.Path
+	if endpoint == "" {
+		endpoint = u.Opaque
+	}
+	if strings.HasPrefix(target, fmt.Sprintf("%s://", u.Scheme)) {
+		endpoint = strings.TrimPrefix(endpoint, "/")
```

Comment:
Was this still necessary? I think we were saying we should just always trim it from `Endpoint` since we never supported `scheme:abs-path` in the past, and we can recommend using `Target.ParsedURL` going forward (maybe also deprecating the old fields)?

Reply:
This might be another pain in the wrong place.

(`dns` is used as a placeholder for a registered scheme, and we don't care about `unix*` schemes since we handle that in the resolver)

- The old behavior is as follows:
`dns:/a/b/c` parses as `{Scheme: defScheme, Endpoint: "dns:/a/b/c"}`
`unregistered:/a/b/c` parses as `{Scheme: defScheme, Endpoint: "unregistered:/a/b/c"}`,
- New behavior with always stripping the leading `/` is as follows:
`dns:/a/b/c` parses as `{Scheme: "dns", Endpoint: "a/b/c"}`
`unregistered:/a/b/c` parses as `{Scheme: defScheme, Endpoint: "unregistered:/a/b/c"}`,
- New behavior with stripping the leading `/` only when `scheme://`:
`dns:/a/b/c` parses as `{Scheme: "dns", Endpoint: "/a/b/c"}`
`unregistered:/a/b/c` parses as `{Scheme: defScheme, Endpoint: "unregistered:/a/b/c"}`,

For an unregistered scheme, the behavior does not change. But for a registered scheme using `scheme:abs-path` was resulting in the default scheme being applied in the past. That will no longer be the case since `url.Parse` thinks that `scheme:/path` and `scheme:///path` are the same.

In the third option above, at least the registered scheme is getting the correct absolute-path. Wdyt?

Reply:
> The old behavior is as follows:
> dns:/a/b/c parses as {Scheme: defScheme, Endpoint: "dns:/a/b/c"}
> unregistered:/a/b/c parses as {Scheme: defScheme, Endpoint: "unregistered:/a/b/c"},

_With_ a custom dialer, it may have been _possible_ for it to produce a working client, but it's quite unusual.  Without a custom dialer, this would have just failed miserably and there's no need to maintain backward compatibility of the method of failure here.

So I'm not sure we care about this case, unless I'm missing something.

Now there is the case where maybe they were dialing using `hostname:port` and _maybe_ their `hostname` aliases to a registered scheme (e.g. the `hostname` they're dialing is the unqualified `unix` or `dns` or `etcd`).  In this case, I think we'll be breaking them no matter what by adding support for `scheme:abs-path` parsing.  This also seems very unlikely and probably acceptable collateral damage in that case.

Reply:
Ack.

Reply:
I think we should always strip to simplify the code / remove special cases, and recommend using the parsed URL and deprecate the old field.  WDYT @menghanl?